builtin/plugins/coin/coin.go

```diff
@@ -76,7 +79,37 @@ func (c *Coin) Meta() (plugin.Meta, error) {
 func (c *Coin) Init(ctx contract.Context, req *InitRequest) error {
 	div := loom.NewBigUIntFromInt(10)
 	div.Exp(div, loom.NewBigUIntFromInt(18), nil)
-
+	//Checks if Coin Policy Feature is enabled before loading Coin Monetary Supply Policy
+	if req.Policy != nil {
+		if req.Policy.DeflationFactorNumerator == 0 {
+			return errors.New("DeflationFactorNumerator should be greater than zero")
+		}
+		if req.Policy.DeflationFactorDenominator == 0 {
+			return errors.New("DeflationFactorDenominator should be greater than zero")
+		}
+		if req.Policy.BaseMintingAmount == 0 {
+			return InvalidBaseMintingAmount
+		}
+		if req.Policy.MintingAccount == nil {
+			return errors.New("Invalid Minting Account Address")
+		}
+		addr := loom.UnmarshalAddressPB(req.Policy.MintingAccount)
+		if addr.Compare(loom.RootAddress(addr.ChainID)) == 0 {
+			return errors.New("Minting Account Address cannot be Root Address")
+		}
+		deflationFactorNumerator := req.Policy.DeflationFactorNumerator
+		deflationFactorDenominator := req.Policy.DeflationFactorDenominator
```

Comment:
What's the point of these temporary vars?

Reply:
Done.